app/controllers/crate/version.js

```diff
@@ -33,6 +33,11 @@ export default Controller.extend({
     isOwner: computed('crate.owner_user', 'session.currentUser.id', function() {
         return this.get('crate.owner_user').findBy('id', this.get('session.currentUser.id'));
     }),
+    notYankedOrIsOwner: computed('model', 'crate.owner_user', 'session.currentUser.id', function() {
```

Comment:
could we add unit tests for this computed?

Reply:
Sure we could, but I'm not familiar with Ember or Mirage, so I'm not sure what's the best way to do this. I've added an attempt as a new commit that tries to use Mirage to set up a new crate and a version. The test passes in its current form, but has a few problems:

* I was only able to use the Mirage factories when running the whole application with `setupApplicationTest()`. This seems unnecessary and makes the test much slower than it should be.
* Testing the other cases (yanked crate, with the user being owner or not) requires also faking an active session and user accounts. I couldn't find any existing infrastructure to do so in a unit test.

We may be able to test this with an acceptance test instead, but at least for me this would involve significant effort. Maybe someone more experienced with Ember could add this test?

Reply:
Apologies @smarnach, I should've added a bit more direction here. Ember can generate default unit tests that should avoid needing to use mirage. The problems you point out highlight the many concerns of acceptance / integration testing I hoped to avoid by suggesting a unit test. I'll open a PR against your branch with a simple unit test.

Reply:
I'm doing this in my fork, but you can create test files using `ember generate` commands. 

```sh
ember generate controller-test crates/version
```

Reply:
I was lazy and didn't fork your copy, but here is the working example of the unit test: https://github.com/rust-lang/crates.io/compare/master...efx:unit-test-controller?expand=1#diff-a30cef8546b1174ebf3c37a550e2357f

Reply:
Thanks a lot, this looks a lot more reasonable than my attempt! I guess I should learn a bit more about Ember before doing any more work on the frontend. :)

I've copied your test to this branch (and force-pushed, since my attempt isn't worth being preserved in the Git history).

Reply:
certainly @smarnach. Eh, learning on the go seems most efficient 😄. Grand!